### PR TITLE
upgrade libbpf to latest

### PR DESCRIPTION
**Pull request by rootfs**
Need to use the `InitGlobalVariable` function in libbpf

The changes include:
- use libbpf go latest release that support libbpf 1.2.0
- adapts to latest libbpf go function signatures
- migrate from bpf to btf maps
- misc changes in Makefile to align `uname -m` and `go arch`

### CODE REVIEW

**Comment by rootfs** [comment]
@sunya-ch I am stuck here with the build error, can you take a look? thanks!

```console
# make _build_local
go mod tidy -v
go mod vendor
./automation/presubmit-tests/gofmt.sh
Checking go format
TAGS=include_gcs include_oss containers_image_openpgp gssapi providerless netgo osusergo gpu libbpf linux
github.com/aquasecurity/libbpfgo
# github.com/aquasecurity/libbpfgo
In file included from vendor/github.com/aquasecurity/libbpfgo/libbpfgo.go:5:
./libbpfgo.h:135:82: error: variable has incomplete type 'enum bpf_cgroup_iter_order'
                                                      enum bpf_cgroup_iter_order order,
                                                                                 ^
./libbpfgo.h:135:60: note: forward declaration of 'enum bpf_cgroup_iter_order'
                                                      enum bpf_cgroup_iter_order order,
                                                           ^
./libbpfgo.h:148:12: error: no member named 'cgroup' in 'union bpf_iter_link_info'
    linfo->cgroup.order = order;
    ~~~~~  ^
./libbpfgo.h:149:12: error: no member named 'cgroup' in 'union bpf_iter_link_info'
    linfo->cgroup.cgroup_fd = cgroup_fd;
    ~~~~~  ^
./libbpfgo.h:150:12: error: no member named 'cgroup' in 'union bpf_iter_link_info'
    linfo->cgroup.cgroup_id = cgroup_id;
    ~~~~~  ^
./libbpfgo.h:151:12: error: no member named 'task' in 'union bpf_iter_link_info'
    linfo->task.tid = tid;
    ~~~~~  ^
./libbpfgo.h:152:12: error: no member named 'task' in 'union bpf_iter_link_info'
    linfo->task.pid = pid;
    ~~~~~  ^
./libbpfgo.h:153:12: error: no member named 'task' in 'union bpf_iter_link_info'
    linfo->task.pid_fd = pid_fd;
    ~~~~~  ^
7 errors generated.
make: *** [Makefile:215: _build_local] Error 2
```

**Comment by rootfs** [comment]
Turns out we need to pick up the `install_uapi_headers` libbpf build target to install headers on the build image.

File: bpfassets/libbpf/src/kepler.bpf.h [hunk 1]
```diff
@@ -59,12 +59,12 @@ typedef __u16 u16;
...
-        .key_size = sizeof(_key_type),                              \
-        .value_size = sizeof(_value_type),                          \
-        .max_entries = _max_entries,                                \
-    };
+    struct {                                                        \
+        __uint(type, _type);                                        \
+        __type(key, _key_type);                             \
+        __type(value, _value_type);                         \
+        __uint(max_entries, _max_entries);                          \
+    } _name SEC(".maps");
```
Comment (rootfs) [inline bpfassets/libbpf/src/kepler.bpf.h:67]: this is needed due to libbpf deprecating bpf maps in favor of btf maps.

File: bpfassets/libbpf/Makefile [hunk 2]
```diff
@@ -7,5 +8,5 @@ BPF_SRC := bpfassets/libbpf/src/*.bpf.c
 $(TARGET_BPF): $(BPF_SRC)
 	clang \
 		-I /usr/include/$(ARCH)-linux-gnu \
-		-O2 -c -target bpf \
-		-o bpfassets/libbpf/bpf.o/$(ARCH)_$@ $<
\ No newline at end of file
+		-O2 -g -c -target bpf \
```
Comment (rootfs) [inline bpfassets/libbpf/Makefile:11]: `-g` is needed for btf

**Comment by rootfs** [comment]
microshift test failures are tracked separately at #826

**Comment by sunya-ch** [comment]
Nice work. The changes look good to me. 
Just need to carefully track the failure on microshift integration test. It seems critical.

**Review by sunya-ch — APPROVED** [review]
should be fine after confirming integration test failure root cause.

**Pull request merged**